File: btap-modeling/test/test_catalog.rb
require_relative 'test_helper'

class TestCatalog < Minitest::Test
  def test_list_all
    rows = BtapModeling.systems
    assert_operator rows.size, :>=, 8
    assert(rows.all? { |r| r['name'] && r['family'] })
  end

  def test_list_filtered
    gas = BtapModeling.systems(filter: 'Gas')
    assert gas.any?
    assert(gas.all? { |r| r['name'].downcase.include?('gas') || r['canonical_name'].downcase.include?('gas') },
           'filter matches legacy or canonical names, case-insensitively')

    psz = BtapModeling.systems(family: 'psz')
    assert psz.any?
    assert(psz.all? { |r| r['family'] == 'psz' })
  end

  def test_resolve_merges_sizing_block
    config = BtapModeling::Catalog.resolve('PSZ RTU Gas and DX Coils and Hot Water Baseboard')
    assert_equal 'psz', config['family']
    assert_equal 'Gas', config['heating_coil_type']
    assert_equal true, config['needs_boiler']
    assert_kind_of Hash, config['sizing']
    assert_equal 'TemperatureDifference', config['sizing']['zone_cooling_design_supply_air_temperature_input_method']
    assert_equal 1.1, config['sizing']['zone_cooling_sizing_factor']
  end

  def test_unknown_name_raises_with_suggestions
    err = assert_raises(ArgumentError) { BtapModeling::Catalog.resolve('PSZ Gas Rooftop Thing') }
    assert_match(/unknown system name/, err.message)
    assert_match(/PSZ RTU/, err.message) # suggestions included
  end
end

File: python/tests/modeling/test_catalog.py
import unittest

import btap.modeling as modeling
from btap.modeling.hvac import catalog


class TestCatalog(unittest.TestCase):
    def test_list_all(self):
        rows = modeling.systems()
        self.assertGreaterEqual(len(rows), 8)
        self.assertTrue(all(r.get('name') and r.get('family') for r in rows))

    def test_list_filtered(self):
        gas = modeling.systems(filter='Gas')
        self.assertTrue(gas)
        self.assertTrue(
            all('gas' in r['name'].lower() or 'gas' in r['canonical_name'].lower()
                for r in gas),
            'filter matches legacy or canonical names, case-insensitively')

        psz = modeling.systems(family='psz')
        self.assertTrue(psz)
        self.assertTrue(all(r['family'] == 'psz' for r in psz))

    def test_resolve_merges_sizing_block(self):
        config = catalog.resolve('PSZ RTU Gas and DX Coils and Hot Water Baseboard')
        self.assertEqual('psz', config['family'])
        self.assertEqual('Gas', config['heating_coil_type'])
        self.assertEqual(True, config['needs_boiler'])
        self.assertIsInstance(config['sizing'], dict)
        self.assertEqual('TemperatureDifference',
                         config['sizing']['zone_cooling_design_supply_air_temperature_input_method'])
        self.assertEqual(1.1, config['sizing']['zone_cooling_sizing_factor'])

    def test_unknown_name_raises_with_suggestions(self):
        with self.assertRaises(ValueError) as ctx:
            catalog.resolve('PSZ Gas Rooftop Thing')
        self.assertRegex(str(ctx.exception), r'unknown system name')
        self.assertRegex(str(ctx.exception), r'PSZ RTU')  # suggestions included


if __name__ == '__main__':
    unittest.main()
